src/core/sys/posix/libgen.d

```diff
@@ -0,0 +1,20 @@
+/**
+ * D header file for POSIX.
+ *
+ * Copyright: 2016 Sociomantic Labs GmbH
+ * License:   <a href="http://www.boost.org/LICENSE_1_0.txt">Boost License 1.0</a>.
+ * Authors:   Leandro Lucarella
+ * Standards: The Open Group Base Specifications Issue 6, IEEE Std 1003.1, 2004 Edition
+ */
+
+/*          Copyright Sociomantic Labs GmbH 2016.
+ * Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ */
+module ocean.stdc.posix.libgen;
+
+extern (C):
```

Comment:
The linux man page says:

>       Both dirname() and basename() may modify the contents of path, so it may be desirable to pass  a  copy when calling one of these functions.

Is this really `pure`? I think adding `pure` might be stretching this too much as different implementations could use static data for example.

Also: http://pubs.opengroup.org/onlinepubs/009695399/functions/basename.html

> The basename() function need not be reentrant. A function that is not required to be reentrant is not required to be thread-safe.

Definitely can't guarantee that all implementations out there are pure.